File: p10.c
#include <stdio.h>

int isprime(int c)
{
    int i;
    if(c%2 == 0)
        return 0;

    for(i = 3; i*i <= c; i += 2)
        if(c%i == 0)
            return 0;

    return 1;
}


int main(void)
{
    int i;
    unsigned long long sum = 2;

    for(i = 3; i < 2000000; i += 2)
        if(isprime(i))
            sum += i;

    printf("sum: %lu\n", sum);
}

File: go/p10.go
package main

import "fmt"

func divides(a, b int) bool {
    return a%b == 0
}

func isPrime(x int) bool {
    if divides(x, 2) {
        return false
    }

    for i := 3; i*i <= x; i += 2 {
        if divides(x, i) {
            return false
        }
    }

    return true
}

func main() {

    var sum int64 = 2
    for i := 3; i < 2000000; i += 2 {
        if isPrime(i) {
            sum += int64(i)
        }
    }

    fmt.Println(sum)
}
